lckernel/cad/builders/cadentity.h

```diff
@@ -142,11 +142,20 @@ namespace lc {
                     return _layer != nullptr;
                 }
 
+                /**
+                 * @brief Throw the appropriate exception if any of the required info is nullptr
+                 */
+                void checkEntityConstraints() const {
+                    if (_layer == nullptr){
```

Comment:
This check is already done in `checkValues` (just above). Can you choose between one of the two methods and remove the other, in order to prevent code duplication ?

Reply:
Ok well so I have two choices -
1) Remove checkEntityConstraints() - in this case I use something like `if(checkValues())` in the build function, however since it returns a bool there would no way to determine what message to send with the runtime exception.

2) Remove checkValues() - will have to update all uses of the checkValues in the code to replace it with a try catch block.
Will also remove it from the qt bridge since it's not being used in the lua code
       ~` .addFunction("checkValues", &lc::builder::LinePatternBuilder::checkValues)`~

I'll go with the second way for now.